docs/monitoring.md

```diff
@@ -0,0 +1,93 @@
+> endpoint will be secured with a self-signed certificate. Since the metrics server is using a self-signed certificate,
+> the Prometheus pod scrape configuration will also require the `insecure_skip_verify` flag set. See
+> [the Prometheus documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tls_config).
+
+## Changing the default Metrics configuration
+
+### Using Helm
+
+If you're using *Helm* to install the NGINX Kubernetes Gateway, set the `metrics.*` parameters to the required values
+for your environment. See the [Helm README](/deploy/helm-chart/README.md).
+
+### Using Manifests
+
+If you're using *Kubernetes manifests* to install NGINX Kubernetes Gateway, you can modify the
+[manifest](/deploy/manifests/nginx-gateway.yaml) to change the default metrics configuration:
+
+#### Disabling metrics
+
+1. Set the `-metrics-disable` [command-line argument](/docs/cli-help.md) to `true` and remove the other `-metrics-*`
+   command line arguments.
+
+2. Remove the metrics port entry from the list of the ports of the NGINX Kubernetes Gateway container in the template
+   of the NGINX Kubernetes Gateway Pod:
+
+    ```yaml
+    - name: metrics
+      containerPort: 9113
+    ```
+
+3. Remove the following annotations from the template of the NGINX Kubernetes Gateway Pod:
+
+    ```yaml
+    annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9113"
+    ```
+
+#### Changing the default port
+
+1. Set the `-metrics-port` [command-line argument](/docs/cli-help.md) to the required value.
+
+2. Change the metrics port entry in the list of the ports of the NGINX Kubernetes Gateway container in the template
+   of the NGINX Kubernetes Gateway Pod:
+
+    ```yaml
+    - name: metrics
+      containerPort: <new-port>
+    ```
+
+3. Change the following annotation in the template of the NGINX Kubernetes Gateway Pod:
+
+    ```yaml
+    annotations:
+        <...>
+        prometheus.io/port: "<new-port>"
+        <...>
+    ```
+
+#### Enable serving metrics via https
+
+1. Set the `-metrics-secure-serving` [command-line argument](/docs/cli-help.md) to `true`.
+
+2. Add the following annotation in the template of the NGINX Kubernetes Gateway Pod:
+
+    ```yaml
+    annotations:
+        <...>
+        prometheus.io/scheme: "https"
+        <...>
+    ```
+
+## Available Metrics
+
+NGINX Kubernetes Gateway exports the following metrics:
+
+- NGINX metrics:
+  - Exported by NGINX. Refer to the [NGINX Prometheus Exporter developer docs](https://github.com/nginxinc/nginx-prometheus-exporter#metrics-for-nginx-oss)
+  - These metrics have the namespace `nginx_kubernetes_gateway`, and include the label `class` which is set to the
+    Gateway class of NKG. For example, `nginx_kubernetes_gateway_connections_accepted{class="nginx"}`.
+
+- [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) metrics. These include:
```

Comment:
Do these metrics have the same namespace as the NGINX metrics?

Reply:
No, they don't. I'm not sure if there's anyway to tweak that either.